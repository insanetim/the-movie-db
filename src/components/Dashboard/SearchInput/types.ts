export interface SearchInputProps {
  query: string
}

export interface SearchInputHookProps extends SearchInputProps {}

export interface SearchInputHook {
  currentValue: string
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  handleSearch: (value: string) => void
}
