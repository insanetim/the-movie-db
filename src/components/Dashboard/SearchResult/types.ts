import {
  ErrorMsg,
  HandlePagination,
  Query,
} from 'src/interfaces/global.interface'
import { IMoviesList } from 'src/interfaces/movie.interface'

export type SearchResultHookProps = SearchResultProps

export type SearchResultHookReturn = {
  error: ErrorMsg
  handlePagination: HandlePagination
  isLoading: boolean
  movies: IMoviesList | undefined
}

export type SearchResultProps = {
  query: Query
}
