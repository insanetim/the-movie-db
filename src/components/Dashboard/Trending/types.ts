import type { IMoviesList } from 'src/interfaces/movie.interface'

export interface TrendingHook {
  error: null | string
  handlePagination: (page: number) => void
  loading: boolean
  movies: IMoviesList | null
}
