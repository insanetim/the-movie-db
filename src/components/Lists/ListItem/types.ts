import { MouseEvent } from 'react'
import { IList } from 'src/interfaces/list.interface'

export type ListItemHookProps = {
  listId: IList['id']
  name: IList['name']
}

export type ListItemHookReturn = {
  handleClick: () => void
  handleListDelete: (event: MouseEvent<HTMLSpanElement>) => () => Promise<void>
}

export type ListItemProps = {
  description: IList['description']
  listId: IList['id']
  name: IList['name']
}
