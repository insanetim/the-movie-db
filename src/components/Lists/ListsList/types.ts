import type { IList } from 'src/interfaces/list.interface'

export interface ListsListProps {
  lists: IList[]
}
