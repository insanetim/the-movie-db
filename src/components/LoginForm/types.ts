import { UserData } from 'src/store/features/auth'

export type LoginFormHookReturn = {
  isDark: boolean
}

export type LoginFormProps = {
  isSubmitting: boolean
  onSubmit: (values: UserData) => void
}
