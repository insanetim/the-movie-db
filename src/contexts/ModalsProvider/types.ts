import { ComponentProps, Dispatch } from 'react'
import { ModalsMap, ModalTypes } from 'src/components/ModalsRoot/types'

export type Modal = {
  [T in ModalTypes]: {
    modalId: string
    modalProps: ComponentProps<ModalsMap[T]>
    modalType: T
  }
}[ModalTypes]

export type ModalsContextType = {
  dispatch: Dispatch<ModalsReducerAction>
  modals: Modal[]
}

export type ModalsReducerAction =
  | CloseModalAction
  | OpenModalAction
  | RemoveModalAction

type CloseModalAction = {
  payload: string
  type: 'CLOSE_MODAL'
}

type OpenModalAction = {
  payload: Modal
  type: 'OPEN_MODAL'
}

type RemoveModalAction = {
  payload: string
  type: 'REMOVE_MODAL'
}
