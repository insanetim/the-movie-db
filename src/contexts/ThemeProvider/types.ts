import { MappingAlgorithm } from 'antd'

export type ThemeProviderHookReturn = {
  algorithm: MappingAlgorithm
  isDark: boolean
}
