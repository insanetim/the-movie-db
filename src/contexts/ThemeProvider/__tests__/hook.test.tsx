import { Theme } from 'src/store/app/types'
import { renderHookWithWrapper } from 'src/utils/testHelpers/renderWithWrapper'

import useContainer from '../hook'

describe('ThemeProvider useContainer hook', () => {
  const mockState = {
    app: {
      _persist: {
        rehydrated: true,
        version: -1,
      },
      modal: {
        modalProps: null,
        modalType: null,
      },
      notifications: [],
      theme: 'light' as Theme,
    },
  }

  it('should match snapshot with light theme', () => {
    const { result } = renderHookWithWrapper(useContainer, {
      preloadedState: mockState,
    })

    expect(result.current).toMatchSnapshot()
  })

  it('should match snapshot with dark theme', () => {
    mockState.app.theme = 'dark' as Theme
    const { result } = renderHookWithWrapper(useContainer, {
      preloadedState: mockState,
    })

    expect(result.current).toMatchSnapshot()
  })
})
