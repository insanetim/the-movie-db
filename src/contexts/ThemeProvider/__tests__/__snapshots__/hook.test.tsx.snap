// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider useContainer hook should match snapshot with dark theme 1`] = `
{
  "algorithm": [Function],
  "isDark": true,
}
`;

exports[`ThemeProvider useContainer hook should match snapshot with light theme 1`] = `
{
  "algorithm": [Function],
  "isDark": false,
}
`;
